src/main/java/com/icthh/xm/tmf/ms/communication/service/mail/MailService.java

```diff
@@ -523,6 +525,10 @@ hasAttachments, true, ofNullable(attachments).map(Map::keySet).orElse(null),
             } else {
                 log.warn("Email could not be sent to user '{}': {}", to, e.getMessage());
             }
+
+            if (applicationProperties.getEmail().isFailOnError()) {
+                throw new IllegalStateException("Email could not be sent to user '" + to + "' : " + e.getMessage());
```

Comment:
how about put original exception to constructor (to keep stacktrace) ?

Reply:
@sergeysenja1992 it will produse unnessary log, stack trace with original now in debug mode

Reply:
@sergeysenja1992 i added it